### PR TITLE
OPS-11861 Setup CircleCI for ef-open

### PR DESCRIPTION
**Pull request by clau-ellation**
**Ready**
[OPS-11861](https://ellation.atlassian.net/browse/OPS-11861)

# Details
This PR sets up CircleCI for ef-open to perform unit testing and pylint for each commit. The CircleCI build flair is also added to the README. Note that we added substantially more pylint checks to disable (e.g., bad-whitespace). Most of these are for coding conventions while a few are for potential errors which have been added to another ticket to investigate.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+version: 2.1
+
+jobs:
+  build-and-test:
+    docker:
+      - image: circleci/python:2.7
+    steps:
+      - checkout
+      - run: sudo pip install .
```
Comment (momoneko) [inline .circleci/config.yml]: Why's this sudo?

Comment (clau-ellation) [reply .circleci/config.yml]: You are right, the whole job ran without sudo -- updated config.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+version: 2.1
+
+jobs:
+  build-and-test:
+    docker:
+      - image: circleci/python:2.7
+    steps:
+      - checkout
+      - run: pip install .
```
Comment (momoneko) [inline .circleci/config.yml]: Recently found out you can install the test dependencies based on `setup.py`'s extras.
```suggestion
      - run: pip install '.[test]'
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      - run: pip install mock pylint
```
Comment (momoneko) [inline .circleci/config.yml]: If we're changing the line above, this one's no longer necessary.
```suggestion
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      - run:
+          command: python -m unittest discover
+          name: Run Unit Test
```
Comment (momoneko) [inline .circleci/config.yml]: Minor nitpick, but do we need the `Run` prefix here, them being run steps and all?
```suggestion
          name: Unit Tests
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      - run:
+          command: pylint --rcfile=./pylintrc --disable=C,W,R ./efopen
+          name: Run Pylint
```
Comment (momoneko) [inline .circleci/config.yml]:
```suggestion
          name: Pylint
```

File: .circleci/config.yml (hunk 2 above)
Comment (clau-ellation) [reply .circleci/config.yml]: Thanks, it worked and cleans up the config.

File: .circleci/config.yml (hunk 3 above)
Comment (clau-ellation) [reply .circleci/config.yml]: Removed.

**Pull request merged**